yt/geometry/oct_visitors.pxd

```diff
@@ -136,3 +142,34 @@ cdef inline int cind(int i, int j, int k) nogil:
     # THIS ONLY WORKS FOR CHILDREN.  It is not general for zones.
     return (((i*2)+j)*2+k)
 
+from oct_container cimport OctreeContainer
+
+cdef class StoreIndex(OctVisitor):
+    cdef np.int64_t[:,:,:,:] cell_inds
+
+# cimport oct_container
+cdef class BaseNeighbourVisitor(OctVisitor):
+    cdef int idim      # 0,1,2 for x,y,z
+    cdef int direction # +1 for +x, -1 for -x
+    cdef np.int8_t[:] neigh_ind
```

Comment:
two quick things -- should `neigh_ind` be a 3d view?  and, it's a bit faster if we declare `::1` in them so that it knows they have stride 1.